core/templates/dev/head/components/gadget_panel.html

```diff
@@ -1,6 +1,6 @@
 <script type="text/ng-template" id="components/gadgetPanel">
-  <div ng-repeat="gadgetSpec in panelContents()">
-    <oppia-gadget class="protractor-test-<[gadgetSpec.gadget_type]>-gadget" gadget-type="gadgetSpec.gadget_type" gadget-customization-args="gadgetSpec.customization_args" show-in-states="gadgetSpec.visible_in_states">
+  <div ng-repeat="gadgetInstance in panelContents()">
+    <oppia-gadget class="protractor-test-<[gadgetInstance.gadget_type]>-gadget" gadget-type="gadgetInstance.gadget_type" gadget-name="gadgetInstance.gadget_name" gadget-customization-args="gadgetInstance.customization_args" show-in-states="gadgetInstance.visible_in_states">
```

Comment:
The problem with using gadget_name is if the name has spaces it ends up breaking into multiple classes. Example: protractor-test-Score Bar-gadget 

Do you know if there's a clever way around this? I touched on this being a limitation of gadget integration testing in the review with Jacob. It's possible the integration tests need to focus on at most one instance of a given gadget type visible on any card.

In prior integration tests it was possible to specify both a gadget_type and panel_name to uniquely identify a gadget instance by its panel, but since a gadget's panel is now hardcoded as part of the gadget we don't have to worry about that use case anymore.

The way I'm thinking about it:
- We catch cases of invalid settings in both front-end and back-end validation (e.g. too many gadgets visible in a panel at the same time)
- Primary role of gadget integration tests may be visual & interactive aspects of specific gadget instances and the ability to add/edit/rename/delete them.